### ISSUE TITLE
Change timing loss function to something depending on the evaluation time

### ISSUE THREAD

**Issue by jacopok** [jacopok/mlgw_bns#31]
The current loss function in `HyperparameterOptimization.objective` doesn't make a lot of sense. 

The validation error part is fine, but the other cost is currently in the form $CN + T$, where $C$ is some constant, $N$ is the number of training points used, while $T$ is the time required to train the network.

This should be replaced by some nonlinear function $f$ of the time $t$ required to evaluate the network. This could simply be evaluated by the number of connections....

**Comment by jacopok** [comment]
It would also be good to explore network complexity "sequentially" somehow, starting with smaller nets and going up as needed.
